scanners/domain-discovery/domain_discovery.py

```diff
@@ -0,0 +1,193 @@
+import subprocess
+import logging
+import os
+from arango import ArangoClient
+from dotenv import load_dotenv
+
+import asyncio
+import nats
+import functools
+import json
+import signal
+import traceback
+
+load_dotenv()
+
+logging.basicConfig(
+    level=logging.INFO, format="[%(asctime)s :: %(name)s :: %(levelname)s] %(message)s"
+)
+logger = logging.getLogger()
+
+DB_USER = os.getenv("DB_USER")
+DB_PASS = os.getenv("DB_PASS")
+DB_NAME = os.getenv("DB_NAME")
+DB_URL = os.getenv("DB_URL")
+
+NAME = os.getenv("NAME", "domain-discovery")
+SUBSCRIBE_TO = os.getenv("SUBSCRIBE_TO")
+PUBLISH_TO = os.getenv("PUBLISH_TO")
+QUEUE_GROUP = os.getenv("QUEUE_GROUP")
+SERVERLIST = os.getenv("NATS_SERVERS")
+SERVERS = SERVERLIST.split(",")
+
+# Establish DB connection
+arango_client = ArangoClient(hosts=DB_URL)
+db = arango_client.db(DB_NAME, username=DB_USER, password=DB_PASS)
+
+
+def subdomain_enumeration(domain):
+    logging.info("Running findomain for {domain}".format(domain=domain))
+    subprocess.run(["findomain", "-t", domain, "-o"])
+
+
+def process_subdomains(base_domain, orgId):
+    f = open("{domain}.txt".format(domain=base_domain), "r")
+    subdomains = f.readlines()
+    subdomains = [domain.strip() for domain in subdomains]
+    claimed_domains = get_claimed_domains(orgId)
+    domains_to_scan = []
+    for subdomain in subdomains:
+        if subdomain not in claimed_domains:
+            logging.info("Adding {subdomain} to DB".format(subdomain=subdomain))
+            checkDomain = db.collection("domains").find({"domain": subdomain}).next()
+            if not checkDomain:
+                domainInsert = db.collection("domains").insert(
+                    {
+                        "domain": subdomain,
+                        "lastRan": None,
+                        "selectors": [],
+                        "hash": None,
+                        "status": {
+                            "certificates": None,
+                            "dkim": None,
+                            "dmarc": None,
+                            "https": None,
+                            "spf": None,
+                            "ssl": None,
+                        },
+                        "archived": False,
+                    }
+                )
+                domains_to_scan.append(domainInsert)
+            else:
+                domainInsert = checkDomain
+            db.collection("claims").insert(
+                {
+                    "_from": orgId,
+                    "_to": domainInsert["_id"],
+                    "hidden": False,
+                    "tags": [],
```

Comment:
Might be worth adding a tag or some other flag for this since it won't be in audit logs.